src/port1.0/portconfigure.tcl

```diff
@@ -77,14 +88,24 @@ proc portconfigure::should_add_stdlib {} {
     return [expr {$has_stdlib && $is_clang}]
 }
 proc portconfigure::should_add_cxx_abi {} {
-    set is_oldos [expr {[option os.platform] eq "darwin" && [option os.major] < 10}]
-    set is_mp_gcc [string match *g++-mp-* [option configure.cxx]]
-    return [expr {$is_oldos && $is_mp_gcc}]
+    # prior to OS X Mavericks, libstdc++ was the default C++ runtime, so
+    #    assume MacPorts libstdc++ must be ABI compatible with system libstdc++
```

Comment:
While they are ABI compatible, that doesn't mean that objects can be passed between them because the runtime is free to maintain its own global state about objects.  Eg:  Consider the case where /usr/lib/libstdc++ allocates an object but it is freed by /opt/local/lib/libstdc++.dylib.  That is one such potential source for problems.

Reply:
I so look forward to the day we flip the switch and default all MacPorts systems to `libc++` and no longer have to worry about much of this.

Reply:
LOL I think we all look forward to that day, Ken!